openembedded-core-mbl/scripts/signctl/README.md

```diff
@@ -0,0 +1,329 @@
+# Mbed Linux OS RPi3 Signing Tool
+
+## Overview
+
+The `signctl` tool automates the basic signing flow for Raspberry PI 3. For more information on the background and details of the signing operations see [the basic signing flow document](https://github.com/ARMmbed/meta-mbl/blob/warrior-dev/docs/basic-signing-flow.md).
+
+The signing tool uses [Hashicorp Vault](https://www.hashicorp.com/products/vault/) to generate and store the certificates required to verify the ATF boot chain.
+
+The Vault secrets model uses 'paths' to 'mount' secrets engines. The 'path' is a REST API endpoint and can be accessed via HTTP requests.
+
+For more background on the Vault secrets engine model see [this link](https://www.vaultproject.io/intro/getting-started/secrets-engines#secrets-engines)
+
+In the Vault PKI engine, each 'mount' of the secrets engine is usually a root CA or intermediate CA, which can then sign and issue certificates from the endpoints created under the base path. Only one CA is allowed per instance of the PKI engine.
+
+The `signctl` tool uses a CA configured at its own mount to represent the private key for each node in the certificate chain of trust.
+The tool then requests leaf certificates from the appropriate mount path, so the corresponding cert is signed by the correct private key.
+
+For more background on the PKI secrets engine see [this link](https://www.vaultproject.io/docs/secrets/pki/index.html#one-ca-certificate-one-secrets-engine).
```

Comment:
Re-word like above to remove "this link" please.

Reply:
Re-worded in bd16b66